ClientApp/viking-vault/src/UserDashboard/CreateAutomaticPaymentModal.tsx

```diff
@@ -0,0 +1,255 @@
+import React from "react";
+import { constants } from "../Resources/Constants.js";
+import '../AdminDashboard/AttachCardModal.css';
+
+const regexCheckIfPositiveFloat = "^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$";
+const currentDate = new Date();
+
+interface IModalProps {
+    open: boolean;
+    onModalClose: any;
+}
+
+interface IAutomaticPaymentProperties {
+    companyName: string;
+    amount: string;
+    initialPaymentDate: string;
+    lastPaymentDate: string;
+}
+
+interface ICompanyData {
+    id: number;
+    name: string;
+    address: string;
+    balance: number;
+}
+
+interface IFormState {
+    [key: string]: string | boolean | IAutomaticPaymentProperties  | ICompanyData[] | null;
+    automaticPayment: IAutomaticPaymentProperties;
+    valid: boolean | null;
+    errorLabel: string;
+    companies: ICompanyData[];
+}
+
+class CreateAutomaticPaymentForm extends React.Component<any, IFormState> {
+    constructor(props: IModalProps) {
+        super(props);
+        this.state = {
+            automaticPayment: {
+                companyName: "",
+                amount: "",
+                initialPaymentDate: "",
+                lastPaymentDate: "",
+            },
+            valid: null,
+            errorLabel: "",
+            companies: [],
+        }
+    }
+
+    private handleChange = (inputValue: string, inputName: string) => {
+        this.setState({
+            automaticPayment: {
+                ...this.state.automaticPayment,
+                [inputName]: inputValue
+            }
+        });
+    }
+
+    private isCompanySelected = () => {
+        if(this.state.automaticPayment.companyName !== "")
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "A company must be selected!",
+            class: "alert alert-info"
+        }
+    }
+
+    private amountMustBePositive = () => {
+        if(this.state.automaticPayment.amount.match(regexCheckIfPositiveFloat))
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
```

Comment:
Here you have 2 returns in the same branch of the if statement. Did you wanna return after the if, maybe? or in the else branch?